app/scripts/modules/core/src/delivery/details/StageFailureMessage.tsx

```diff
@@ -54,14 +74,14 @@ export class StageFailureMessage extends React.Component<IStageFailureMessagePro
 
   public render() {
     const { message, messages } = this.props;
-    const { isFailed, failedTask, failedStage, failedStageIndex } = this.state;
+    const { isFailed, failedTask, failedExecutionId, failedStage, failedStageIndex, failedStepIndex } = this.state;
     if (isFailed || failedTask || message || messages.length) {
       const exceptionTitle = messages.length ? 'Exceptions' : 'Exception';
       const displayMessages = message || !messages.length ?
         <p>{message || 'No reason provided.'}</p> :
         messages.map((m, i) => <p key={i}>{m || 'No reason provided.'}</p>);
 
-      if (failedTask || !failedStage) {
+      if (message || messages.length) {
```

Comment:
If we have messages, display them.

If we do not have messages but we have coordinates to a failure, display a link (either to a stage in the same execution, or a stage in a child execution).

Otherwise, show nothing (ie. no longer show the `No reason provided.`).